test/space/ntuples_test.py

```diff
@@ -346,112 +347,76 @@ def test_power(fn):
     assert all_almost_equal(y_pos, y_pos_arr)
 
 
-def _test_unary_operator(fn, function):
-    # Verify that the statement y=function(x) gives equivalent results
-    # to NumPy
-    x_arr, x = example_vectors(fn)
+def test_unary_ops(fn):
+    # Verify that the unary operators (`+x` and `-x`) work as expected
 
-    y_arr = function(x_arr)
+    for op in [operator.pos, operator.neg]:
+        x_arr, x = example_vectors(fn)
 
-    y = function(x)
+        y_arr = op(x_arr)
+        y = op(x)
 
-    assert all_almost_equal([x, y], [x_arr, y_arr])
+        assert all_almost_equal([x, y], [x_arr, y_arr])
 
 
-def _test_binary_operator(fn, function):
-    # Verify that the statement z=function(x,y) gives equivalent results
+def _test_scalar_operator(fn, function):
+    # Verify that the statement y=op(x, scalar) gives equivalent results
     # to NumPy
+
+    for scalar in [-31.2, -1, 0, 1, 2.13]:
+        x_arr, x = example_vectors(fn)
+
+        if scalar == 0 and function in [operator.truediv,
+                                        operator.itruediv]:
+            # Check for correct zero division behaviour
+            with pytest.raises(ZeroDivisionError):
+                y = function(x, scalar)
+        else:
+            y_arr = function(x_arr, scalar)
+            y = function(x, scalar)
+
+            assert all_almost_equal([x, y], [x_arr, y_arr])
+
+        # right op
+        x_arr, x = example_vectors(fn)
+
+        y_arr = function(scalar, x_arr)
+        y = function(scalar, x)
+
+        assert all_almost_equal([x, y], [x_arr, y_arr])
```

Comment:
~~If we assume that `function` wouldn't work as expected and not write to its `out` parameter (which should be used as keyword arg BTW), then the test would pass since before, `x` and `x_arr` had the same value, and they're both not changed.~~
Okay, so now I finally realized that `function` comes from a fixture by another layer of indirection, and that it uses the magic methods once on the Numpy array and once on the ODL data structure. Either I'm stupid or this could very well use an explaining comment. Otherwise it's going to be magic to anyone not willing to spend 15 minutes on figuring out what's going on.